pkg/controller/controller_instance.go

```diff
@@ -273,24 +273,11 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1alpha1.ServiceIn
 		request.OriginatingIdentity = originatingIdentity
 	}
 
-	// If the instance is failed, just clear the finalizer
-	if isServiceInstanceFailed(instance) {
-		glog.V(5).Infof("Clearing catalog finalizer from ServiceInstance %v/%v", instance.Namespace, instance.Name)
-
-		// Clear the finalizer
-		finalizers.Delete(v1alpha1.FinalizerServiceCatalog)
-		toUpdate.Finalizers = finalizers.List()
-
-		if _, err = c.updateServiceInstanceStatus(toUpdate); err != nil {
-			return err
-		}
-
-		return nil
+	if instance.Status.OrphanMitigationInProgress && instance.Status.OperationStartTime == nil {
+		now := metav1.Now()
+		toUpdate.Status.OperationStartTime = &now
```

Comment:
Need to update the status in the API registry at this point, prior to sending the request to the Broker.

Reply:
In this case, I was thinking that since the "Provision" operation was still ongoing, and we were only updating immediately when the CurrentOperation changed, that we wouldn't need to update here.

Reply:
I think I'm changing my thinking on `CurrentOperation`. I think its scope should only be what immediate broker call you've made. So we would, for instance, set `ServiceInstanceOperationProvision` before we message the broker, and then clear it at the end of that reconciliation, no matter what the result was. This is unlike today, where on an error that we retry, or an async response, we keep the CurrentOperation in place.

Reply:
I think the status should show the operation as "provision" during orphan mitigation.